### PR TITLE
Fix PPoly integrate and test for CubicSpline

### PR DESCRIPTION
**Pull request by dmadeka**
Fixes which variable should become readonly.

Also fixes read only issues with `PPoly.integrate` @ev-br

### CODE REVIEW

**Comment by dmadeka** [comment]
@ev-br This might be all we can do till the fused types const fixes are made for Cython

**Comment by dmadeka** [comment]
@tylerjereddy Id add the defect label to this too; no reason PPoly shouldnt take read-only memviews

File: scipy/interpolate/tests/test_interpolate.py [hunk 1]
```diff
@@ -1012,7 +1012,7 @@ def test_read_only(self):
         p = PPoly(c, x, extrapolate='periodic')
 
         for writeable in (True, False):
-            xnew.flags.writeable = writeable
+            x.flags.writeable = writeable
```
Comment (ev-br) [inline scipy/interpolate/tests/test_interpolate.py]: I'd make the both readonly.

File: scipy/interpolate/tests/test_interpolate.py [hunk 2]
```diff
@@ -1279,6 +1279,19 @@ def test_integrate(self):
...
+
+        for writeable in (True, False):
+            x.flags.writeable = writeable
+
+            P = PPoly(c, x)
+            I = P.antiderivative()
+
+            vals = I(4) - I(1)
+            assert_(np.isfinite(vals).all())
+
```
Comment (ev-br) [inline scipy/interpolate/tests/test_interpolate.py]: This does not test the integrate method IIRC.

File: scipy/interpolate/tests/test_interpolate.py (hunk 1 above)
Comment (dmadeka) [reply scipy/interpolate/tests/test_interpolate.py]: I set xnew to True and False, and x to False. Does that sound good?

File: scipy/interpolate/tests/test_interpolate.py (hunk 2 above)
Comment (dmadeka) [reply scipy/interpolate/tests/test_interpolate.py]: You're right - fixed to call `integrate` directly

**Comment by ev-br** [comment]
Thanks @dmadeka , it's in.

**Pull request merged**